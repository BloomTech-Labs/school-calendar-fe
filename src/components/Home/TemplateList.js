import React, {useState} from 'react'
import Template from './Template'






//brings up modal for user to generate a new template and add to list of preexisting ones. 
const addTemplate = () => {
  console.log('a modal will pop when the button is clicked')
}


//grabs users list of existing templates, and populated the field. 
const TemplateList = () => {

  //place holder list to be replaced by axios call
  const [list, setList] = useState([
    {
      id:1,
      name:'basketball'
    },
    {
      id:2,
      name:'math class'
    },
    {
      id:3,
      name:'beer drinking'
    }
  ])


  return (
    <div>
      {list.map(temp => {
        return( <Template
          key={temp.id}
          thing={temp.name}
        />)
      })}
      <button onClick={() => addTemplate()}> + New Template </button>
    </div>
  )
}

export default TemplateList
