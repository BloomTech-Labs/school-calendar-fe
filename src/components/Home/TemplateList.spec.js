import React from 'react';
import { shallow } from 'enzyme';
import TemplateList from './TemplateList';



describe('TemplateList', () => {
    it('should render correctly', () => {
      const component = shallow(<TemplateList />);
    
      expect(component).toMatchSnapshot();
    });
  
    
  });