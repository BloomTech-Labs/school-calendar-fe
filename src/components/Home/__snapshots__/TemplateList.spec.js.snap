// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateList should render correctly 1`] = `
<div>
  <Template
    key="1"
    thing="basketball"
  />
  <Template
    key="2"
    thing="math class"
  />
  <Template
    key="3"
    thing="beer drinking"
  />
  <button
    onClick={[Function]}
  >
     + New Template 
  </button>
</div>
`;
